add unavailable status with an example (#22607)
home-assistant/home-assistant.github.io: source/_docs/configuration/templating.markdown

- Iterating `states` will yield each state sorted alphabetically by entity ID.
- Iterating `states.domain` will yield each state of that domain sorted alphabetically by entity ID.
- `states.sensor.temperature` returns the state object for `sensor.temperature` (avoid when possible, see note below).

> - `states('device_tracker.paulus')` will return the state string (not the object) of the given entity, `unknown` if it doesn't exist, `unavailable` if the object exists but not yet available (e.g. due to an integration internal processing) 

Reviewer: What does "due to an integration internal processing" mean?

Author: Hello,
The root cause of this PR is because, I've got some error in templating with error such as 'None Type'.
When I look deeper in HomeAssistant source code, I've seen that for templating, I got "3 states" : 
- `STATE_UNKNOWN` (const.py)
- `STATE_UNAVAILABLE` (const.py)
- the native state value

Unknown was quite clear (e.g. 'none')
Unavailable can be set by the integration via the method available (or _attr_available attribute) (cf. `helpers/entity.py`)
That's why I put  

> integration internal processing

If you have another proposal, I take

Reviewer: My proposal would be:

```suggestion
- `states('device_tracker.paulus')` will return the state string (not the object) of the given entity, `unknown` if it doesn't exist, `unavailable` if the object exists but is not yet available.
```

Reasoning: I have no idea what internal processing means. `unavailable` has nothing to do with Home Assistant internals.

Author: I upate

  Paulus is at {{ states('device_tracker.paulus') }}.
{% endif %}

#check state and compute remaining time before a train departure time (in minutes)
{% if (states('sensor.train_departure_time') != "unknown" and states('sensor.train_departure_time') != "unavailable") %}
{{ ( (as_timestamp(states('sensor.train_departure_time')) - as_timestamp(now()))/60.0 ) | round (2) }}
{% else %}
unknown

> {% endif %}

Reviewer: This looks like an extremely overcomplicated example that isn't really helpful in this location IMHO.

What is it you are trying to explain with this example?

Author: You're right ...
I just wanted to show an example with a test with unknown/unavailable state. Is the following example better ?
```
#check state and make a processing
{% if (states('sensor.train_departure_time') != "unknown" and states('sensor.train_departure_time') != "unavailable") %}
{{ ... }}
{% else %}
unknown
{% endif %}
```

NB: due to unavailable state , the jinja statement 'is defined' does not work (indeed the variable exists)

Reviewer: Why are you transforming `unavailable` into `unknown` in this template? I'm not following that.

Anyways, it can also be written as `{% if states('sensor.train_departure_time') in ("unavailable", "unknown") %}`

That said, `unknown` should not be needed to return at all (as that is a so called `None` value), so the whole `else` part can be dropped too.

Author: I upate
